ui/src/containers/AppsTableContainer.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { sortable, SortByDirection } from '@patternfly/react-table';
+import AppsTable from '../components/AppsTable';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { getApps } from '../actions/actions';
+
+class AppsTableContainer extends React.Component {
+  constructor (props) {
+    super(props);
+    this.sortBy = {};
+    this.onSort = this.onSort.bind(this);
+    this.onRowClick = this.onRowClick.bind(this);
+  }
+  componentDidMount () {
+    this.props.getApps();
+  }
+  onRowClick (event, rowId, props) {
+    this.setState({ redirect: true });
+  }
+  onSort (_event, index, direction) {
+    const sortedRows = this.props.apps.sort((a, b) => (a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0));
+    this.setState({
+      sortBy: {
+        index,
+        direction
+      },
+      apps: direction === SortByDirection.asc ? sortedRows : sortedRows.reverse()
+    });
+  }
+
+  render () {
+    const columns = [
```

Comment:
@craicoverflow Good question. My initial thought is that only data should be stored in state. I do need to investigate this further to see if we should store values of this kind, i.e. strings which are shown on the UI which won't ever need to be mutated. Open to thoughts on this one.

Reply:
My opinion would be that the state object stores the state of the component. The Patternfly React [example](http://patternfly-react.surge.sh/patternfly-4/components/table) in their docs does this:

```jsx
class SortableTable extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      columns: [
        { title: 'Repositories', transforms: [sortable] },
        'Branches',
        { title: 'Pull requests', transforms: [sortable] },
        'Workspaces', 'Last Commit'
      ],
      rows: [
        ['one', 'two', 'a', 'four', 'five'],
        ['a', 'two', 'k', 'four', 'five'],
        ['p', 'two', 'b', 'four', 'five']
      ],
      sortBy: {}
    };
    this.onSort = this.onSort.bind(this);
  }

```

Reply:
@craicoverflow thanks for looking into this. Does that mean we should store all component state regardless of whether it's data that would be stored/fetched from the backend? So we'd store both app data and non-app data in state. I had them separated in my mind by open to clarification here - I was relying on my memory from doing Redux ~3 years ago!

Reply:
The code you see here is based on the example that you shared, the only difference is that it's split out to container/component and the data is being fetched via Redux.

Reply:
@laurafitzgerald to be honest I don't know. Don't have any knowledge of Redux so don't know how it should work with that, but when not using Redux then yes I think it should be stored in the state regardless of where it is fetched from - if it affects the state of the component then it is in the state object. I don't have enough knowledge on this to be certain about what I am saying though.

Reply:
Ok. I've pushed some of the testing work so will look into this a little more while I wait for a review on that. Thanks @craicoverflow 😄

Reply:
agree with @craicoverflow and think component state should definitely be stored in the redux store. Data should also I think (as we are doing), with my thinking being if we were not using redux we would be storing the data in the component's state with `setState()` so as avoid unnecessary RESTful calls. Stored data for our app should not be excessive either, so storing it in redux store will be sufficient.

Reply:
This can be resolved as part of my follow on pr or we need to create a JIRA. Posted about it in the chat so waiting on a reply there and resolving this conversation.